src/bin/cargo/commands/read_manifest.rs

```diff
@@ -1,7 +1,10 @@
+//! Deprecated.
+
 use crate::command_prelude::*;
 
 pub fn cli() -> Command {
     subcommand("read-manifest")
+        .hide(true)
```

Comment:
This is the only commit I'm unsure about.

On one hand, this will reduce clutter and make completions nicer.

On the other hand, `--list` is no longer exhaustive which can be important for knowing what command names are available.  Unsure what other use cases might be affected.

Reply:
That reminds me https://github.com/rust-lang/cargo/issues/12114.

`cargo --list` may be a little brittle to change. I've restored the behavior of `--list` and only hide commands for a nicer shell completion.
Also add some DEPRECATED/REMOVED prefix for making the status more standing out in `cargo --list`.